src/link/Elf/Object.zig

```diff
@@ -105,12 +105,12 @@ fn parseCommon(self: *Object, allocator: Allocator, handle: std.fs.File, elf_fil
     defer allocator.free(header_buffer);
     self.header = @as(*align(1) const elf.Elf64_Ehdr, @ptrCast(header_buffer)).*;
 
-    const target = elf_file.base.comp.root_mod.resolved_target.result;
-    if (target.cpu.arch != self.header.?.e_machine.toTargetCpuArch().?) {
+    const em = elf_file.base.comp.root_mod.resolved_target.result.toElfMachine();
+    if (em != self.header.?.e_machine) {
         try elf_file.reportParseError2(
             self.index,
-            "invalid cpu architecture: {s}",
-            .{@tagName(self.header.?.e_machine.toTargetCpuArch().?)},
+            "invalid ELF machine type: {s}",
+            .{@tagName(self.header.?.e_machine)},
         );
         return error.InvalidCpuArch;
```

Comment:
Since we are changing the meaning of the error, it would be good to update the error too from `error.InvalidCpuArch` to `error.InvalidElfMachineType`.

Reply:
I went with `InvalidMachineType` (without `Elf`) because this error is used across COFF/ELF/Mach-O code, so it seemed better to keep it standardized. Is that ok?